Applications/Spire/Source/KeyBindings/TaskKeysTableView.cpp

```diff
@@ -430,10 +411,34 @@ struct UniqueTaskKeyTableModel : TableModel {
   }
 
   QValidator::State set(int row, int column, const std::any& value) override {
-    return m_source->set(row, column, value);
+    auto update_key = [&] (const Region& region, const QKeySequence& key) {
```

Comment:
When a series of operations need to be performed that are all related to one another, use a transaction to avoid introducing inconsistent states.